sycl/include/CL/sycl.hpp

```diff
@@ -38,8 +38,7 @@
 
 // These require > C++ 17 for now, but compiling all the tests with C++17 breaks
 // some Clang Unit tests involving SYCL making it difficult to test for breakage.
-#if (defined(__cplusplus) && (__cplusplus == 201703L))
-#define __SYCL_VENDOR_XILINX_EXTENSIONS__
+#if (defined(__cplusplus) && (__cplusplus >= 201703L))
 #include <CL/sycl/xilinx/fpga.hpp>
```

Comment:
I am a little bit worried by the end-user interface.
For example if you have a program using a Xilinx FPGA extension but an old C++ version, it will fail because the extension does not exist, instead of the compilation to fail because of the Xilinx extension is asking for some future stuff.

Reply:
I think in reality the check can be removed and we just state that we only support compilation for C++17/C++2a, that will be the inevitable end result anyway as we further modify the run-time and the changes leak outside of the Xilinx extensions.

So just be explicit about not supporting earlier C++ versions. 

The alternative I can see would be to create multiple C++ interfaces that are optionally included based on the C++ implementation I suppose. Which would have to be done for anything we extend to support newer features, but that will start to look like what's already done here and doesn't seem like the ideal solution after some initial tinkering.

Reply:
The only real issue in doing this, is that it's going to get hard to check what tests are failing unless I go through all the SYCL tests and up their C++ std. Which I suppose will be inevitable, but cause some merge conflicts at some point I imagine.

Reply:
So I agree, I think this will/should be another PR